numpy/lib/function_base.py

```diff
@@ -3670,17 +3883,90 @@ def _quantile_is_valid(q):
     return True
 
 
-def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
-                           interpolation='linear', keepdims=False):
+def _validate_weights(a, q, axis, weights):
+    if weights is None:
+        return None
+    else:
```

Comment:
I'd skip this else, which allows you to remove a level of indentation from everything below it